DEFAULT_CONFIG.json5

```diff
@@ -462,11 +462,14 @@
   ///
   //
   //  plugins_loading: {
-  //    // Enable plugins loading.
+  //    /// Enable plugins loading.
   //    enabled: false,
   //    /// Directories where plugins configured by name should be looked for. Plugins configured by __path__ are not subject to lookup.
-  //    /// If `enabled: true` and `search_dirs` is not specified then `search_dirs` falls back to the default value: ".:~/.zenoh/lib:/opt/homebrew/lib:/usr/local/lib:/usr/lib"
-  //    search_dirs: [],
+  //    /// Directories are specified as object with fields `kind` and `value` is accepted.
+  //    ///   1. If `kind` is `current_exe_parent`, then the parent of the current executable's directory is searched and `value` should be `null`.
+  //    ///   2. If `kind` is `path`, then `value` is interpreted as a filesystem path. Simply supplying a string instead of a object is equivalent to this.
+  //    /// If `enabled: true` and `search_dirs` is not specified then `search_dirs` falls back to the default value:
+  //    search_dirs: [{ "kind": "current_exe_parent" }, ".", "~/.zenoh/lib", "/opt/homebrew/lib", "/usr/local/lib", "/usr/lib"],
```

Comment:
What's hence the difference from `{ "kind": "current_exe_parent" }` and `"."`?

Reply:
If I'm running from a Bash shell `zenohd` then:

- `{ "kind": "current_exe_parent" }` equals `$(which zenohd)`
- `"."` equals `$PWD`

As soon as `zenohd` is not in my working directory the two are different.

Reply:
Understood, then should we be simply using something like `__exec_folder__` instead of the whole `{ "kind": "current_exe_parent" }`. In other parts of the config we are using `__<x>__` to mark reserved/special config parameters.

Reply:
Technically it is a valid possibility in JSON. But how would you express it in YAML? There is some discussion on potentially having YAML as default config in Zenoh and I'd like to avoid JSON-specific way of constructing config fields which are not so ergonomic in YAML (or any other data format we may want to support).

Reply:
```yaml
# config.yml
plugins_loading:
  enabled: true
  search_dirs:
    - { kind: current_exe_parent }
    - .
    # Example of a custom field, equivalent to '~/source/zenoh/target/debug'
    - { kind: path, value: ~/source/zenoh/target/debug }
    - ~/.zenoh/lib
    - /opt/homebrew/lib
    - /usr/local/lib
    - /usr/local
```

`yq -o=json < config.yml` yields:

```json
{
  "plugins_loading": {
    "enabled": true,
    "search_dirs": [
      {
        "kind": "current_exe_parent"
      },
      ".",
      {
        "kind": "path",
        "value": "~/source/zenoh/target/debug"
      },
      "~/.zenoh/lib",
      "/opt/homebrew/lib",
      "/usr/local/lib",
      "/usr/local"
    ]
  }
}
```

Even better, [YAML version 1.2 is a superset of JSON](https://en.wikipedia.org/wiki/YAML).

Reply:
Having something like `- { kind: path, value: ~/source/zenoh/target/debug }` looks pretty weird in YAML to be honest. I would expect to have something:
```yaml
# config.yml
plugins_loading:
  enabled: true
  search_dirs:
    - .
    - ~/foo
    # equivalent to
    - path: .
    - path: ~/foo
    # getting information from the environment
    - env: current_exec_folder
```

Which yields the following JSON: 
```json
{
    "plugins_loading": {
        "enabled": true,
        "search_dirs": [
            {
                "path": "."
            },
            {
                "path": "~/foo"
            },
            {
                "env": "current_exec_folder"
            },
            ".",
            "~/foo"
        ]
    }
}
```

Reply:
In any case, the user is free not to use them, at the cost of an extra line in the config.

However, the `{ k: v }` syntax or "inline blocks" have been there since YAML 1.0. [You find them 6 examples into the specification.](https://yaml.org/spec/1.0/#id2489726)

They're useful when an object is small enough to span one line:

```yaml
build:
  - { target: x86_64-unknown-linux-gnu, os: ubuntu-20.04, cross: false }
  - { target: x86_64-unknown-linux-musl, os: ubuntu-20.04, cross: true }
  - { target: arm-unknown-linux-gnueabi, os: ubuntu-20.04, cross: true }
  - { target: arm-unknown-linux-gnueabihf, os: ubuntu-20.04, cross: true }
  - { target: armv7-unknown-linux-gnueabihf, os: ubuntu-20.04, cross: true }
  - { target: aarch64-unknown-linux-gnu, os: ubuntu-20.04, cross: true }
  - { target: aarch64-unknown-linux-musl, os: ubuntu-20.04, cross: true }
  - { target: x86_64-apple-darwin, os: macos-12, cross: false }
  - { target: aarch64-apple-darwin, os: macos-12, cross: false }
  - { target: x86_64-pc-windows-msvc, os: windows-2019, cross: false }
  - { target: x86_64-pc-windows-gnu, os: windows-2019, cross: false }
```

In the above example, without inline blocks the section would be 3x times bigger, going from 11 lines to 33 (not to mention the extra indentation), for no good reason.

They are syntactically very similar to inline arrays (e.g. `[x, y, z]`). Although inline arrays are more common than inline blocks in small configurations.

Reply:
Good point. The very last thing I'd simply add the equivalence of paths in the comment in the config. I.e.:
```
{ "kind": "current_exe_parent" } equals basename $(which zenohd)
"." equals $PWD
```